### PR TITLE
common compilation for Riviera-PRO

### PR DESCRIPTION
**Pull request by dawidzim**
Hey!
In this pull request I added support for common compilation in Riviera-PRO. This compilation mode will be used as default and to change mode we need to type `--compilation_mode=separate`

### CODE REVIEW

File: src/cmd/script.rs [hunk 1]
```diff
@@ -104,6 +104,18 @@ pub fn new<'a, 'b>() -> App<'a, 'b> {
                 .long("no-abort-on-error")
                 .help("Do not abort analysis/compilation on first caught error (only for programs that support early aborting)")
         )
+        .arg(
+            Arg::with_name("compilation_mode")
+                .long("compilation_mode")
```
Comment (fabianschuiki) [inline src/cmd/script.rs]: How about changing the long name to `compilation-mode` to stay in line with the other multi-word switches?

Comment (dawidzim) [reply src/cmd/script.rs]: Ok, I changed long name as you suggest

**Comment by dawidzim** [comment]
So I used `cargo fmt` and I combined commits to one and I think now is fine

**Comment by fabianschuiki** [comment]
Perfect, thanks! Merging.

**Pull request merged**